### PR TITLE
[#38] [Backend] As a user, I can submit my survey response

### LINKED ISSUE #38: [Backend] As a user, I can submit my survey response
## Why
After filling in the answer to all questions, we need to submit them to our end

## Acceptance Criteria
Call `POST /api/v1/responses` endpoint with the following body:

```
{
  "survey_id": "{{survey-id}}",
  "questions": [
      {
          "id": "{{answer-id}}",
          "answer": "{{answer}}" // optional
     },
     {
           "id": "{{answer-id}}",
           "answer": "{{answer}}" // optional
     }
  ]
}
```

## Resources
[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#8de38274-3f9f-4b27-8122-58d7399b4c69)

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/38

## What happened 👀

- Add `submitSurvey()` to `SurveyService` and `SurveyRepository`
- Add `SubmitSurveyUseCase`
- Write Unit Tests

## Insight 📝

n/a

## Proof Of Work 📹

CleanShot 2023-05-10 at 10 48 15

### CODE REVIEW

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package co.nimblehq.mark.kmmic.data.service.survey.model
+
+import co.nimblehq.mark.kmmic.domain.model.SurveySubmission
+import co.nimblehq.mark.kmmic.domain.model.SurveySubmissionAnswer
+import co.nimblehq.mark.kmmic.domain.model.SurveySubmissionQuestion
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+internal data class SurveySubmissionParams(
```
Comment (suho) [inline shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt:10]: Is there any change with and without `internal`?

Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt:10]: We won't access this class from the native side, I aim to public the use case and models only 🙏

**Review by AVI5HEK — APPROVED** [review]
LGTM

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt (hunk 1 above)
Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt:10]: Can you double check, on the iOS side, we can access those internal classes 🤔

Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt:10]: @suho 
CleanShot 2023-05-10 at 14 49 03

Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/survey/model/SurveySubmissionParams.kt:10]: Cool! Thanks for the checking 🙌

**Review by suho — APPROVED** [review]

**Review by phongvhd93 — APPROVED** [review]

**Review by phongvhd93 — APPROVED** [review]

**Pull request merged**